build-info-extractor-gradle/src/main/groovy/org/jfrog/gradle/plugin/artifactory/task/helper/TaskHelperPublications.java

```diff
@@ -248,13 +248,21 @@ public boolean hasModules() {
                     addMavenArtifactToDeployDetails(deployDetails, publicationName, builder, artifactInfo, mavenPublication);
                 }
             }
-            // Second adding the main artifact of the publication, if present
-            if (mavenNormalizedPublication.getMainArtifact() != null) {
-                createPublishArtifactInfoAndAddToDeployDetails(mavenNormalizedPublication.getMainArtifact(), deployDetails, mavenPublication, publicationName);
+
+            Set<MavenArtifact> artifacts;
+            try {
+                // Gradle 5.0 and above:
+                artifacts = mavenNormalizedPublication.getAdditionalArtifacts();
+                // Second adding the main artifact of the publication, if present
+                if (mavenNormalizedPublication.getMainArtifact() != null) {
+                    createPublishArtifactInfoAndAddToDeployDetails(mavenNormalizedPublication.getMainArtifact(), deployDetails, mavenPublication, publicationName);
+                }
+            } catch (NoSuchMethodError error) {
+                // Compatibility with older versions of Gradle:
+                artifacts = mavenNormalizedPublication.getAllArtifacts();
```

Comment:
This collection will again contain the Maven POM artifact.
So it needs to be filtered again in the code below, especially as this plugin has an option to not publish the POM file.